aws/config.go

```diff
@@ -458,6 +459,20 @@ func (c *Config) Client() (interface{}, error) {
 		}
 	}
 
+	// Infer URL Suffix from STS Endpoint
+	resolver := endpoints.DefaultResolver()
+	endpoint, err := resolver.EndpointFor(endpoints.StatesServiceID, client.region)
```

Comment:
The comment notes inferring this information from the STS endpoint, however its using the states service.

That said, the AWS Go SDK documentation lists the `XXXServiceID` constants as deprecated: https://docs.aws.amazon.com/sdk-for-go/api/aws/endpoints/#pkg-constants

> Deprecated: Use client package's EndpointsID value instead of these ServiceIDs. These IDs are not maintained, and are out of date.

This should be swapped out according to that deprecation.

Ideally though, we should fetch this information directly from the AWS Go SDK endpoints information first, then only fallback to parsing an endpoint should that fail since endpoint hostnames may not always follow a consistent implementation. I have submitted an upstream feature request in the AWS to directly provide the DNS suffix information, which can help prevent us from hardcoding these cases in the future: https://github.com/aws/aws-sdk-go/issues/2710

Reply:
This one I'm not sure when I'll have time to research and figure out the alternative. If anyone wants to take it, please be my guest!